core/src/test/scala/shapeless/annotation.scala

```diff
@@ -175,4 +188,30 @@ class AnnotationTests {
     illTyped(" TypeAnnotations[Second, Dummy] ", "could not find implicit value for parameter annotations: .*")
   }
 
+  @Test
+  def allAnnotations: Unit = {
+    {
+      val all: (First :: HNil) :: shapeless.HNil :: (Second :: Third :: HNil) :: HNil = AllAnnotations[CC3].apply()
+      assert(all == (First() :: HNil) :: shapeless.HNil :: (Second(2, "b") :: Third('c') :: HNil) :: HNil)
+    }
+
+    {
+      val first = AllAnnotations[CC3].apply()
+      assert(first == (First() :: HNil) :: shapeless.HNil :: (Second(2, "b") :: Third('c') :: HNil) :: HNil)
+    }
+  }
+  
+  @Test
+  def allTypeAnnotations: Unit = {
+    {
+      val all: (First :: HNil) :: shapeless.HNil :: (Second :: Third :: HNil) :: HNil = AllTypeAnnotations[CC4].apply()
```

Comment:
You can use `typed` instead of testing this twice

Reply:
same as previous